### ISSUE TITLE
Integration of EthStamp Model Score Endpoint for Passport

### ISSUE THREAD

**Issue by erichfi** [gitcoinco/passport#2034]
### User Story:
As a Passport integrator,
I want to access an endpoint that can return the EthStamp model score for any Ethereum address,
So that I can implement nuanced access control in my app based on reliable scoring data, without necessitating users to have a Passport or undergo specific stamp verifications.

### Acceptance Criteria
GIVEN I am an integrator with a valid API key from the scorer app,
WHEN I make a request to the EthModel scoring endpoint with an Ethereum address,
THEN I should receive a response containing a numeric integer score between 0 and 100 for that address, indicating the likelihood of the address being operated by a human versus a Sybil entity, with a guarantee that a score will be provided for every Ethereum mainnet account queried.

#### Product & Design Links:
Endpoints for MVP:
- `/registry/score/eth-activity/{address}` - Queries a single specified Ethereum address.
- `/registry/score/eth-activity` - Queries multiple specified addresses through a parameter.

Special Parameters:
- `address` (for single and multiple address queries), `pagination`.

Example Response:
For an Ethereum address with its calculated score:

```
{
  "address": "0x8dabc053c7fda425826d913a58c0dc4c2a0b36d7",
  "eth_activity_score": "56.337828000",
  "timestamp": "2024-01-31T16:02:32.767754+00:00"
}
```
Key Details:
The system is capable of returning a calculated score for any account on the Ethereum mainnet, eliminating the need for the user to interact with Passport directly or verify specific stamps. This ensures broad applicability and utility for integrators seeking to implement nuanced access controls based on the EthStamp model score.

#### Tech Details:
- The implementation requires the development of Identity Provider (IDP) endpoints for authentication and rate limiting.
- The scorer and the passport & model API are currently in separate Virtual Private Clouds (VPCs), necessitating strategic planning for secure and efficient communication between these services.

#### Notes/Assumptions:

- The system is designed to return a calculated score for any Ethereum mainnet address queried, ensuring broad applicability for integrators.
- There will always be a calculated score returned for each wallet, ensuring consistent and reliable data for access control decisions.

**Comment by nutrina** [comment]
This will be a bit tricky: we'll need do expose an IDP endpoint for the authentication & rate limiting part, as the scorer is not yet in the same VPC as the passport & model API.
We'll need a follow-up ticket to remedy this as soon as the migration to the same VPC is done

**Comment by nutrina** [comment]
This depends on #2037

**Comment by michellema1208** [comment]
gm @erichfi ! A couple q's from Grants Stack side: 

- Do users need to verify the Ethereum stamp on Passport in order for Passport integrators (i.e. Grants Stack) to be able to pull in EthModel scoring2? Or will there be a way to query the EthModel endpoint for a score given a wallet address, even if that user has not verified with the Ethereum stamp on Passport? 
- Will there be data/scores returned for users that don't fall into the ETH Advocate/Pioneer/Maxi buckets? Where can we find more info/definition about those buckets? 

Also lmk if this is the best medium for these q's -- happy to move them off Github and into a Discord thread if that's better!

**Comment by lebraat** [comment]
Some DevRel suggestions for what I believe we should provide around this use case:


First off, this is assuming that a user doesn't need to submit a Passport via the POST endpoint to retrieve data.

I recommend we (eventually) make three endpoints available, with the first (and maybe second) being our MVP:

/registry/score/eth-activity/{address} -- Single specified address
/registry/score/eth-activity -- multiple specified addresses via a parameter
* Special Parameters = address, pagination, 
/registry/score/eth-activity/{scorer_id) -- Retrieve all eth activity scores that have been submitted to a scorer
* Special Parameters = pagination

Example response with verified ETH stamp:
{
	"address": "0x8dabc053c7fda425826d913a58c0dc4c2a0b36d7",
	"eth_activity_score": "56.337828000",
	"timestamp": "2024-01-31T16:02:32.767754+00:00",
}

Example response without verified ETH stamp:
{
	"address": "0x8dabc053c7fda425826d913a58c0dc4c2a0b36d7",
	"eth_activity_score": "null",
	"timestamp": "2024-01-31T16:02:32.767754+00:00",
}


Questions: 
* Will we be able to eth_activity_score Passports that have not verified the ETH stamps?
* Will we be able to eth_activity_score wallets that don't have a Passport?



NOT MVP
Eventually, we should add the following parameters that expands the initial object with additional data:
* include_humanity_score=true/false
* include_binary_humanity_score=true/false
* include_verified_stamps=all,kyc,{specific_stamp or stamps_comma-separated}
* include_deduped_data=true/false

**Comment by lebraat** [comment]
Update looks great!!

Question:
How quick will we be able to score wallets with this? Will we be able to handle ~100 wallet addresses and still deliver the data in a timely fashion?

And also looking forward, Are we going to have a different field for each model? 

Ex/

{
"address": "0x8dabc053c7fda425826d913a58c0dc4c2a0b36d7",
"overall_activity_score": "75.23058000"
"models": {
    "eth_activity_score": "56.337828000",
    "nft_activity_score": "79.23058000",
    "base_activity_score": "null"
    },
"timestamp": "2024-01-31T16:02:32.767754+00:00",
}

**Comment by lebraat** [comment]
Thanks for the detailed response, as always!

This ticket's scope is great for the MVP! Excited to work with you all on further evolving this to work with additional models!

**Comment by digitalmnt** [comment]
The linked PR is a WIP implementation of invoking the existing eth model lambda from within the scorer api. This way we can still use the existing rate limiting implemented in the scorer api. I'm seeing requests come through when testing against the staging lambda, but the input is incorrectly formatted. I added comments to the PR

**Comment by lucianHymer** [comment]
Thanks for all the awesome input @lebraat !

This last paragraph is basically what I was thinking about this morning:
```
NOT MVP
Eventually, we should add the following parameters that expands the initial object with additional data:

    include_humanity_score=true/false
    include_binary_humanity_score=true/false
    include_verified_stamps=all,kyc,{specific_stamp or stamps_comma-separated}
    include_deduped_data=true/false
```

I almost think this should be flipped around so that we have e.g. a /registry/passport/{address} endpoint that returns details about the user and optionally rolls everything up into a top-level pass/fail.

```
{
  "address": "0xabc",
  "pass": true,
  "details": {
    "models": {
      "eth_stamp_model": {
        "score": "78"
      }
    }
  }
}
```

Eventually, the options (which models to use, built in credentials to include, custom credentials to include, whether to include our recommended pass/fail, pass/fail logic to use, block-lists to include (custom or Passport-curated), allow-lists to include, etc.) could be configured on a scorer-by-scorer basis in the scorer UI.

This is basically the same thing, just framed a little differently (i.e. a /passport endpoint instead of an /eth_activity endpoint).

I just think it'll be an uphill battle to get people to switch from the /eth_activity endpoint to passport-lite. I think we should just go ahead and frame it as an alpha passport lite, with the defaults set to:

- Only include eth_stamp_model
- Don't assess pass/fail

So basically, matching the requirements for this issue.

Then as we add more functionality, we could change the defaults for _new_ scorers and allow the existing scorers to enable these options if they want.

Also, I don't know what integrators would do with a bunch of different, maybe conflicting, model scores. If I'm an integrator and I see `"eth_stamp_model": 70, "nft_model": 0`, what do I do with that? I'm not 100% sure how exactly we roll this stuff up to a pass/fail in the final version, but I don't think integrators would have any idea how to do that either. I think that's something we should figure out and prescribe.  Although if an integrator does want to implement custom logic for this, they're free to do so and we'll give them all the necessary details.

I see this eventually replacing `submit_passport`, and we just automatically run models/web3 stamps whenever necessary and cache the values (e.g. the models only need to be run once per month per user and this value can be shared across all scorers). If an integrator wants everything to be entirely stamp-based, they can enable stamps and disable the models.

A later version of the response from this endpoint might look more like

```
{
  "address": "0xabc",
  "pass": true,
  "details": {
    "blocklists": {
      "know-sybils": {
        "version": "2.3",
        "included": false
      }
    },
    "allowlists": {
      "dao-leaders": {
        "version": "6.1",
        "included": false,
      }
    },
    "models": {
      "eth_stamp_model: {
        version: "1.2",
        score: "78",
      }
      "nft_model":  {
        version: "0.8",
        score: "50",
      }
    },
    "credentials": {
      "TwitterAccountGTE180": {
        "deduped": false,
        "score": 2.3
      }
    },
    "passFailMechanism": {
      "name": "PassportRecommendedStrict",
      "version": "1.5",
    }
  }
}
```

**Comment by lucianHymer** [comment]
To review with other engineer before completing
